esmvalcore/cmor/check.py

```diff
@@ -57,11 +57,16 @@ def __init__(self,
                  var_info,
                  frequency=None,
                  fail_on_error=False,
+                 raise_exception=True,
```

Comment:
One extra argument should be sufficient, e.g.
`raise_on=` with possible values `'never', 'warning', 'error'`

Reply:
see my comment above about dry-run